test/e2e/framework.go

```diff
@@ -133,6 +135,25 @@ func NewFramework(t ginkgo.GinkgoTInterface) *Framework {
 	}
 }
 
+// NamespacedTest generates a unique namespace name and sets up
+// a Before/AfterEach to create and delete the namespace before/after the
+// test body. Intended to be used to wrap a single test.
+func (f *Framework) NamespacedTest(body func(string)) func() {
```

Comment:
I definitely like the wrapper, but I think I prefer non-random names, makes it a little easier if debugging tests locally to find what you're looking for, especially if we're running things concurrently.

Reply:
Also some tests use >1 namespace since they're testing cross-namespace things, so those could either have custom handling or you could allow `NamespacedTest` to set up/tear down multiple namespaces